docker/sds-api/spine-directory-service/sds/request/cpm.py

```diff
@@ -170,6 +173,68 @@ def validate_filters(self, query_parts):
         if non_empty_count < 2:
             self._raise_invalid_query_params_error()
     
+    def set_mhs_endpoint(self, ldap_results):
+        for key, ldap_result in enumerate(ldap_results):
+            service, interaction = self._extract_service_and_interaction(ldap_result['nhsMhsSvcIA'])
+            if service in RELIABLE_SERVICES:
+                address: Optional[str] = None
+                for core_spine_interaction, interactions in INTERACTION_MAPPINGS.items():
+                    if interaction in interactions:
+                        address = self._get_interaction_address(core_spine_interaction)
+                        break
+
+                if address:
+                    ldap_results[key]['nhsMHSEndPoint'] = address
+                
+        return ldap_results
+    
+    def _get_interaction_address(self, interaction: str):
+        forward_address_cache: Optional[str] = None
+        forward_address_cache = forward_address_cache or self._get_address(interaction)
+        return forward_address_cache
```

Comment:
the first part of the ternary can never be true, so i'd just simplify to

```suggestion
        return self._get_address(interaction)
```